### PR TITLE
Extend syntax with setting [priority] of task

### PR DESCRIPTION
**Pull request by greeneng**
I extended syntax of Todo plugin for my project (but probably from older version).
Features: add 1, 2 or 3 exclamation marks (!) into <todo> tag to set priority.
When rendered this option just adds a class to <span class="todotext ..."> 
Example 1: <todo !>low priority task</todo>
Example 2: <todo !!!>high priority task</todo>

### CODE REVIEW

**Review by einhirn — CHANGES_REQUESTED** [review]
Please address the comments I added to the code.

**Comment by einhirn** [comment]
Do you still want this to be merged? Please address the requested changes and resolve the conflicts. Otherwise I'll probably close this PR in 1-3 Months or so...

**Comment by greeneng** [comment]
I've updated files in pull request

File: syntax/todo.php [hunk 1]
```diff
@@ -167,14 +164,7 @@ public function render($mode, Doku_Renderer $renderer, $data) {
...
                 return true;
             }
-
-        } elseif($mode == 'metadata') {
-            /** @var $renderer Doku_Renderer_metadata */
-            if($state == DOKU_LEXER_UNMATCHED) {
-                $id = $this->_composePageid($todotitle);
-                $renderer->internallink($id, $todotitle);
-            }
-        }
```
[the remark this answers is not in the document]
Comment (einhirn) [reply syntax/todo.php:171]: Probably just because the PR currently still is based on our code 12 commits ago...

**Comment by einhirn** [comment]
@greeneng Please rebase your pull request onto our main branch. In your own git repo do the following to clean up your git log of non-descript commits and get your fork up to date with this repo in one go:
```
git add remote upstream https://github.com/leibler/dokuwiki-plugin-todo.git
git fetch upstream
git checkout master
git rebase --interactive upstream/main
vi/nano/whatever <conflicted file>
git add <conflicted file>
git rebase --continue
# rinse repeat
``` 

It will probably fail with conflicts, but it will tell you what to do. Of course you'll need to adjust the conflicted files so your new code is actually added to the already existing code...
There are helpful manuals like this out there:
https://docs.github.com/en/get-started/using-git/about-git-rebase

and it links to pages about what to do when there are conflicts during a rebase:
https://docs.github.com/en/get-started/using-git/resolving-merge-conflicts-after-a-git-rebase
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-using-the-command-line

**Pull request closed without merging**